engine/pkg/rpcutil/client_test.go

```diff
@@ -124,3 +124,27 @@ func TestFailToDialLeaderAfterwards(t *testing.T) {
 	require.ErrorIs(t, err, derror.ErrNoRPCClient)
 	t.Log(err.Error())
 }
+
+func getMockTimeoutDail(temporary bool) DialFunc[*mockRPCClient] {
+	retryCount := 0
+	return func(_ context.Context, addr string) (*mockRPCClient, CloseableConnIface, error) {
+		if addr != "timeout" {
+			return &mockRPCClient{addr: addr}, &closer{}, nil
+		}
+		retryCount++
+		if temporary && retryCount == defaultDialRetry {
```

Comment:
The name of `temporary` is not easy to understand

Reply:
Rename it to `succEventually`, PTAL